org.eclipse.epp.mpc.core/src/org/eclipse/epp/internal/mpc/core/service/xml/StripInvalidXMLCharsReader.java

```diff
@@ -25,31 +25,34 @@ public StripInvalidXMLCharsReader(Reader in) {
 	@Override
 	public int read() throws IOException {
 		for (int r = -1; (r = super.read()) != -1;) {
-			if (isValidXMLCodePoint(r)) {
+			if (isValidXMLChar((char) r)) {
 				return r;
 			}
 		}
 		return -1;
 	}
 
-	private boolean isValidXMLCodePoint(int cp) {
-		return (cp == 0x9) || (cp == 0xA) || (cp == 0xD) || ((cp >= 0x20) && (cp <= 0xD7FF))
-				|| ((cp >= 0xE000) && (cp <= 0xFFFD)) || ((cp >= 0x10000) && (cp <= 0x10FFFF));
+	private boolean isValidXMLChar(char c) {
+		// https://marketplace.eclipse.org/content/surfexample/api/p
+		// Strip ASCII control characters disallowed in general by XML 1.0.
+		return (c >= 0x20) || (c == 0x9) || (c == 0xA) || (c == 0xD);
```

Comment:
Yes, there might be invalid unpaired or improperly pair surrogates, but filtering those is way too complicated, and that was not even the original problem being fixed.